docs/src/main/asciidoc/stork-registration.adoc

```diff
@@ -0,0 +1,139 @@
+- The application registers itself automatically using the detected or configured IP address and port.
+- During shutdown, the service is automatically deregistered.
+
+The solution is located in the `stork-automatic-consul-registration-quickstart` link:{quickstarts-tree-url}/stork-automatic-consul-registration-quickstart[directory].
+
+
+== Use cases/configuration options
+
+The registration system uses sensible defaults but also allows for fine-grained control per service.
+
+=== Case 1: implicit configuration
+
+No configuration required at all:
+
+[source,xml]
+----
+<dependency>
+  <groupId>io.smallrye.stork</groupId>
+  <artifactId>stork-service-registration-consul</artifactId>
+</dependency>
+----
+
+Quarkus automatically uses:
+
+- Application name
+- Detected IP address
+- HTTP port
+
+The service is registered and unregistered automatically.
+
+=== Case 2: minimal explicit configuration
+
+Only override what's necessary:
+
+[source,properties]
+----
+quarkus.stork.my-service.service-registrar.ip-address=192.168.0.42
+----
+
+No `quarkus.stork.my-service.service-registrar.type` is needed because there's only one registrar configuration in `application.properties`.
+
+=== Case 3: multiple services registrars
+
+When multiple services are configured, you must specify the `quarkus.stork.<service-name>.service-registrar.typetype`:
+
+[source,properties]
+----
+quarkus.stork.red-service.service-registrar.type=consul
+quarkus.stork.red-service.service-registrar.port=8083
+
+quarkus.stork.blue-service.service-registrar.type=consul
+quarkus.stork.blue-service.service-registrar.port=8084
+----
+
+=== Case 4: disabling registration
+
+Automatic registration is enabled by default. If you want to disable registration, you can do it per service using the following property:
+
+[source,properties]
+----
+quarkus.stork.red-service.service-registrar.enabled=false
+----
+
+=== Case 5: advanced health check configuration
+
+Customize how Consul polls service health:
+
+[source,properties]
+----
+quarkus.stork.my-service.service-registrar.parameters.health-check-url=/q/health/live
+quarkus.stork.my-service.service-registrar.parameters.health-check-interval=10s
+quarkus.stork.my-service.service-registrar.parameters.health-check-deregister-after=1m
+----
+
+=== Case 6: static IP and port
+
+Useful when behind a proxy or in production with known IPs:
+
+[source,properties]
+----
+quarkus.stork.my-service.service-registrar.ip-address=203.0.113.25
+quarkus.stork.my-service.service-registrar.port=80
+----
+
+== IP detection
+
+By default, Quarkus attempts to determine the IP address using the first available, non-loopback network interface.
+If this fails, it falls back to `localhost`.
+
+Note that in Docker/Kubernetes environments, this detection may result in internal IPs that are not externally reachable.
+In such cases, it's recommended to explicitly set the IP and port.
```

Comment:
There's no way to determine a valid IP/port from the environment in the Docker/Kubernetes case?